x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsqlExpressionTranslators.java

```diff
@@ -415,4 +421,60 @@ static Query translate(
             }
         }
     }
+
+    public static class Ranges extends ExpressionTranslator<Range> {
+
+        @Override
+        protected Query asQuery(Range r, TranslatorHandler handler) {
+            return doTranslate(r, handler);
+        }
+
+        public static Query doTranslate(Range r, TranslatorHandler handler) {
+            return handler.wrapFunctionQuery(r, r.value(), () -> translate(r, handler));
+        }
+
+        private static RangeQuery translate(Range r, TranslatorHandler handler) {
+            Object lower = valueOf(r.lower());
+            Object upper = valueOf(r.upper());
+            String format = null;
+
+            // for a date constant comparison, we need to use a format for the date, to make sure that the format is the same
+            // no matter the timezone provided by the user
```

Comment:
It was removed indeed https://github.com/elastic/elasticsearch/pull/102767
Probably some of this logic for now is redundant

Reply:
There is indeed some redundant logic here.

We should probably only move this into ESQL if it's also cleaned up, otherwise it looks like it's intended this way. 

I'll remove the migration part of this PR, to only add the tests instead.